source/common/router/rds_impl.cc

```diff
@@ -119,5 +125,56 @@ void RdsRouteConfigProviderImpl::registerInitTarget(Init::Manager& init_manager)
   init_manager.registerTarget(*this);
 }
 
+RouteConfigProviderManagerImpl::RouteConfigProviderManagerImpl(
+    Runtime::Loader& runtime, Event::Dispatcher& dispatcher, Runtime::RandomGenerator& random,
+    const LocalInfo::LocalInfo& local_info, ThreadLocal::SlotAllocator& tls)
+    : runtime_(runtime), dispatcher_(dispatcher), random_(random), local_info_(local_info),
+      tls_(tls) {}
+
+std::vector<Router::RouteConfigProviderSharedPtr>
+RouteConfigProviderManagerImpl::routeConfigProviders() {
+  std::vector<Router::RouteConfigProviderSharedPtr> ret;
+  ret.reserve(route_config_providers_.size());
+  for (const auto& element : route_config_providers_) {
+    // Because the RouteConfigProviderManager's weak_ptrs only get cleaned up
+    // in the RdsRouteConfigProviderImpl destructor, and the single threaded nature
+    // of this code, locking the weak_ptr will not fail.
+    Router::RouteConfigProviderSharedPtr provider = element.second.lock();
+    ASSERT(provider)
+    ret.push_back(std::move(provider));
+  }
+  return ret;
+};
+
+Router::RouteConfigProviderSharedPtr RouteConfigProviderManagerImpl::getRouteConfigProvider(
+    const Json::Object& config, Upstream::ClusterManager& cm, Stats::Scope& scope,
+    const std::string& stat_prefix, Init::Manager& init_manager) {
+
+  // RdsRouteConfigProviders are unique based on their <route_config_name>_<cluster>.
+  const std::string manager_identifier =
+      config.getString("route_config_name") + ":" + config.getString("cluster");
```

Comment:
The key is opaque. If you just remove ':' and then you can allow ':' in the name (you probably could allow ':' in the name anyway and it would work).